vmprof/reader.py

```diff
@@ -43,12 +71,12 @@ class JittedCode(int):
     pass
 
 def wrap_kind(kind, pc):
-    if kind == VMPROF_ASSEMBLER_TAG:
-        return AssemblerCode(pc)
-    elif kind == VMPROF_JITTED_TAG:
-        return JittedCode(pc)
-    assert kind == VMPROF_CODE_TAG
-    return pc
+    cls = {
+        VMPROF_ASSEMBLER_TAG: AssemblerCode,
+        VMPROF_JITTED_TAG: JittedCode,
+        VMPROF_CODE_TAG: lambda x: x,
+    }[kind]
```

Comment:
Microbenchmarks of `wrap_kind`:

```
In [4]: %timeit wrap1(VMPROF_ASSEMBLER_TAG, 1)
The slowest run took 10.88 times longer than the fastest. This could mean that an intermediate result is being cached
1000000 loops, best of 3: 391 ns per loop

In [5]: %timeit wrap2(VMPROF_ASSEMBLER_TAG, 1)
The slowest run took 8.83 times longer than the fastest. This could mean that an intermediate result is being cached
1000000 loops, best of 3: 788 ns per loop

In [6]: %timeit wrap1(VMPROF_JITTED_TAG, 1)
The slowest run took 9.87 times longer than the fastest. This could mean that an intermediate result is being cached
1000000 loops, best of 3: 457 ns per loop

In [7]: %timeit wrap2(VMPROF_JITTED_TAG, 1)
The slowest run took 9.24 times longer than the fastest. This could mean that an intermediate result is being cached
1000000 loops, best of 3: 784 ns per loop

In [8]: %timeit wrap1(VMPROF_CODE_TAG, 1)
The slowest run took 11.83 times longer than the fastest. This could mean that an intermediate result is being cached
1000000 loops, best of 3: 220 ns per loop

In [9]: %timeit wrap2(VMPROF_CODE_TAG, 1)
The slowest run took 8.46 times longer than the fastest. This could mean that an intermediate result is being cached
1000000 loops, best of 3: 691 ns per loop
```

So it's an increase of max. 500 ns per call; compared to I/O cost it really does not matter.